fenics_ice/inout.py

```diff
@@ -304,39 +305,97 @@ def field_from_vel_file(infile, field_name):
 
     return np.ravel(field[:])
 
-def read_vel_obs(params, model=None):
+def read_vel_obs(infile, model=None, use_cloud_point=False):
     """
-    Read velocity observations & uncertainty from HDF5 file
-
-    For now, expects an HDF5 file
+    Reads velocity observations & uncertainty from a HDF5 file
+    containing:
+    - point cloud velocities (u_obs, v_obs) to compute the value
+        of the cost function only and only used if use_cloud_point=True.
+    - gridded data set of composite ice velocities & uncertainties
+        use for alpha initialisation and boundary conditions even
+        if point cloud data is given. And also use to compute the value
+        of the cost function as default.
+    All variables in the HDF5 file should be with the form
+    (values, )
+
+    Params:
+    -------
+    infile: path to .h5 path
+    model: fenics_ice model object
+    use_cloud_point: bool
+    Returns:
+    --------
+    model.vel_obs: as a python directory with the velocity data
+    out: python directory if model=None
     """
-    infile = Path(params.io.input_dir) / params.obs.vel_file
     assert infile.exists(), f"Couldn't find velocity observations file: {infile}"
 
     infile = h5py.File(infile, 'r')
 
-    x_obs = field_from_vel_file(infile, 'x')
-    y_obs = field_from_vel_file(infile, 'y')
+    # Read composite mean of velocity data as default
+    mask_vel = field_from_vel_file(infile, 'mask_vel')
+    x = field_from_vel_file(infile, 'x')
+    y = field_from_vel_file(infile, 'y')
     u_obs = field_from_vel_file(infile, 'u_obs')
     v_obs = field_from_vel_file(infile, 'v_obs')
     u_std = field_from_vel_file(infile, 'u_std')
     v_std = field_from_vel_file(infile, 'v_std')
-    mask_vel = field_from_vel_file(infile, 'mask_vel')
-
-    assert x_obs.size == y_obs.size == u_obs.size == v_obs.size
-    assert v_obs.size == u_std.size == v_std.size == mask_vel.size
 
-    uv_obs_pts = np.vstack((x_obs, y_obs)).T
+    if use_cloud_point:
+        # Read cloud point observations to be used in the Inversion and
+        # Cost function optimization only!
+        x_cloud = field_from_vel_file(infile, 'x_cloud')
+        y_cloud = field_from_vel_file(infile, 'y_cloud')
+        u_cloud = field_from_vel_file(infile, 'u_cloud')
+        v_cloud = field_from_vel_file(infile, 'v_cloud')
+        u_cloud_std = field_from_vel_file(infile, 'u_cloud_std')
+        v_cloud_std = field_from_vel_file(infile, 'v_cloud_std')
+    else:
+        x_cloud = x.copy()
+        y_cloud = y.copy()
+        u_cloud = u_obs.copy()
+        v_cloud = v_obs.copy()
+        u_cloud_std = u_std.copy()
+        v_cloud_std = v_std.copy()
+
+    uv_cloud_pts = np.vstack((x_cloud, y_cloud)).T
+    uv_obs_pts = np.vstack((x, y)).T
+
+    out = defaultdict(list)
+    out['uv_obs_pts'].append(uv_cloud_pts)
+    out['u_obs'].append(u_cloud)
```

Comment:
@jrmaddison  I cant name this cloud in the model.vel_obs map because of the input data use on the test... I dont know how Joe generated that data set and his variables are always name with keys = _obs ... so I can only change the new var names that I have generated.

Reply:
Is `append` used (instead of assignment) so that we can later store multiple sets of observations?

Reply:
Hi I remember now why append is use ... at first we have an empty dictionary and we append one new variable at the time... if I just use = or assign (which does not exist on list) I cant do it because the object that I am appending is not yet there ... so I append a new variable at the time to the dictionary

Reply:
If we don't need multiple datasets then we can use a dictionary
```
out = {}
out['uv_obs_pts'] = uv_cloud_pts
```
This also avoids `out[missing_key]` leading to unexpected behaviour later (as this would create an empty list).

Reply:
ah  I see what you mean now :)  Will change that then and commit asap many thanks!